src/modules/CSADigitizer/CSADigitizerModule.cpp

```diff
@@ -0,0 +1,337 @@
+    config_.setDefault<int>("output_plots_scale", Units::get(30, "ke"));
+    config_.setDefault<int>("output_plots_bins", 100);
+
+
+    // Copy some variables from configuration to avoid lookups:
+    tmax_ = config_.get<double>("amp_time_window");
+    if (config_.get<bool>("simple_parametrisation")){
+      tauF_ = config_.get<double>("feedback_time_constant");
+      tauR_ = config_.get<double>("rise_time_constant");
+      cf_ = config_.get<double>("feedback_capacitance");
+      rf_=tauF_/cf_; 
+      LOG(DEBUG) << "Parameters: cf " << Units::display(cf_/1e-15, "C/V")  << ", rf " << Units::display(rf_, "V*s/C")
+		 << ", tauF_ " << Units::display(tauF_, "s") << ", tauR_ " << Units::display(tauR_, "s");
+    }
+    else{
+      ikrum_ = config_.get<double>("krummenacher_current");
+      cd_ = config_.get<double>("detector_capacitance");
+      cf_ = config_.get<double>("feedback_capacitance");
+      co_ = config_.get<double>("amp_output_capacitance");
+      gm_ = config_.get<double>("transconductance"); 
+      vt_ = config_.get<double>("v_temperature");
+
+      // helper variables: transconductance and resistance in the feedback loop 
+      // weak inversion: gf = I/(n V_t) (e.g. Binkley "Tradeoff and Optimisation in Analog CMOS design")
+      // n_wi typically 1.5, for circuit descriped in  Kleczek 2016 JINST11 C12001: I->I_krumm/2
+      gf_ = ikrum_/(2.0*1.5*vt_)  ;  
+      rf_ = 2./gf_;       //feedback resistor
+      tauF_=rf_*cf_;
+      tauR_=(cd_*co_)/(gm_ *cf_);
+      LOG(DEBUG) << "Parameters: rf " << Units::display(rf_, "V*s/C") << ", cf_ " << Units::display(cf_, "C/V")
+		 << ", cd_ " << Units::display(cd_, "C/V") << ", co_ " << Units::display(co_, "C/V")
+		 << ", gm_ " << Units::display(gm_, "C/s/V") << ", tauF_ " << Units::display(tauF_, "s")
+		 << ", tauR_ " << Units::display(tauR_, "s");      
+    }
+    
+    output_plots_ = config_.get<bool>("output_plots");
+    output_pulsegraphs_ = config_.get<bool>("output_pulsegraphs");
+
+
+    // impulse response of the CSA from Kleczek 2016 JINST11 C12001
+    // H(s) = Rf / ((1+ tau_f s) * (1 + tau_r s)), with
+    // tau_f = Rf Cf , rise time constant tau_r = (C_det * C_out) / ( gm_ * C_F )
+    // inverse Laplace transform R/((1+a*s)*(1+s*b)) is (wolfram alpha) (R (e^(-t/a) - e^(-t/b))) /(a - b) 
+    fImpulseResponse_ = new TF1("fImpulseResponse_", "[0] * ( exp(-x / [1]) - exp(-x/[2]) ) / ([1]-[2])", 0, tmax_);
+    fImpulseResponse_->SetParameters(rf_, tauF_, tauR_);
+
+
+}
+
+
+
+void CSADigitizerModule::init() {
+
+    if(config_.get<bool>("output_plots")) {
+        LOG(TRACE) << "Creating output plots";
+
+        // Plot axis are in kilo electrons - convert from framework units!
+        int maximum = static_cast<int>(Units::convert(config_.get<int>("output_plots_scale"), "ke"));
+        auto nbins = config_.get<int>("output_plots_bins");
+
+	//asv do these histos make sense? what about one with the noise?
```

Comment:
General comment: please do use `make format` to properly format your code. Otherwise the CI on GitLab will fail immediately. Actually, CMake should print a warning when running, suggesting you to install git hooks which run the formatting check automatically before you can commit.